Clear/main.py

```diff
@@ -0,0 +1,335 @@
+
+
+def run_command(command, pager=False):
+    if pager is True:
+        click.echo(click.style("Command: ", fg='cyan') + click.style(command, fg='green'))
+        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+        click.echo_via_pager(p.stdout.read())
+    else:
+        click.echo(click.style("Command: ", fg='cyan') + click.style(command, fg='green'))
+        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+        click.echo(p.stdout.read())
+
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help', '-?'])
+
+
+#
+# 'cli' group (root group) ###
+#
+
+# THis is our entrypoint - the main "Clear" command
+@click.group(cls=AliasedGroup, context_settings=CONTEXT_SETTINGS)
+def cli():
+    """SONiC command line - 'Clear' command"""
+    pass
+
+
+#
+# 'ip' group ###
+#
+
+# This allows us to add commands to both cli and ip groups, allowing for
+# "Clear <command>" and "Clear ip <command>" to function the same
+@cli.group()
+def ip():
+    """Clear IP """
+    click.echo("Command incomplete")
+    pass
+
+#
+# 'bgp' group ####
+#
+
+# We use the "click.group()" decorator because we want to add this group
+# to more than one group, which we do using the "add_command() methods below.
+@click.group(cls=AliasedGroup,default_if_no_args=True)
+def bgp():
+    """Clear BGP (Border Gateway Protocol) peers"""
+    pass
+
+
+# Add 'bgp' group to both the root 'cli' group and the 'ip' subgroup
+cli.add_command(bgp)
+ip.add_command(bgp)
+
+
+# Default 'bgp' command (called if no subcommands or their aliases were passed)
+@bgp.command(default=True)
+def default():
+    """Clear all BGP peers"""
+    command = 'sudo vtysh -c "clear ip bgp *"'
+    run_command(command)
+
+@bgp.group(cls=AliasedGroup, default_if_no_args=True, context_settings=CONTEXT_SETTINGS)
+def neighbor():
+    pass
+
+bgp.add_command(neighbor)
+
+@neighbor.command(default=True)
+@click.argument('ipaddress', required=False)
+def default(ipaddress):
+    """Clear BGP neighbors"""
+    if ipaddress is not None:
+	command = 'sudo vtysh -c "clear ip bgp {} "'.format(ipaddress)
+    else:
+	command = 'sudo vtysh -c "clear ip bgp *"'
+    run_command(command)
+
+# 'in' subcommand
+@neighbor.command('in')
+@click.argument('ipaddress', required=False)
+def neigh_in(ipaddress):
+	if ipaddress is not None:
+		command = 'sudo vtysh -c "clear ip bgp {} in"'.format(ipaddress)
+	else:
+		command = 'sudo vtysh -c "clear ip bgp * in"'
+        run_command(command)
+
+# 'out' subcommand
+@neighbor.command('out')
+@click.argument('ipaddress', required=False)
+def neigh_out(ipaddress):
+	if ipaddress is not None:
+	        command = 'sudo vtysh -c "clear ip bgp {} out"'.format(ipaddress)
+	else:
+		command = 'sudo vtysh -c "clear ip bgp * out"'
+        run_command(command)
+
+@neighbor.group(cls=AliasedGroup, default_if_no_args=True, context_settings=CONTEXT_SETTINGS)
+def soft():
+    pass
+
+neighbor.add_command(soft)
```

Comment:
This line is unnecessary, as neighbor is added to the `neighbor` group via the decorator on line 179.